(WIP) Cli node flags

PlayMa256 opened the pull request:
<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->

**What kind of change does this PR introduce?**
<!-- E.g. a bugfix, feature, refactoring, build related change, etc… -->

**Did you add tests for your changes?**

**If relevant, did you update the documentation?**

**Summary**

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**
<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

**Other information**

ev1stensberg commented on bin/webpack.js:
```diff
@@ -1,515 +1,35 @@
-	Author Tobias Koppers @sokra
-*/
-
-(function() {
```
Don't remove IIFEE here, its for perf + being able to use return.

ev1stensberg commented on bin/webpack.js:
```diff
@@ -1,515 +1,35 @@
-	// Prefer the local installation of webpack-cli
-	if (importLocal(__filename)) {
-		return;
+const spawn = require("cross-spawn");
```
Is everything from here taken from the previous PR? #377

PlayMa256 commented on bin/process-node-options.js:
```diff
@@ -0,0 +1,44 @@
+	});
+
+	// when main process was exited.
+	process.on("exit", (code) => {
```
This was the part "missing" @ev1stensberg , the other one above just relied on `Ctrl/Cmd+C` which sometimes was not the case. I've compared twice master branch and this branch, both are running almost identical (spawning 1 process, when tests for node-flags are running) in terms of process being spawned and this branch is not spawning billions anymore.

PlayMa256 commented:
It is failing, but it isnt even starting to run the tests, it is failing with the following error message: `Object.entries is not a function`: 
```
npm install --global codecov
C:\Users\appveyor\AppData\Roaming\npm\codecov -> C:\Users\appveyor\AppData\Roaming\npm\node_modules\codecov\bin\codecov
+ codecov@3.0.2
added 55 packages from 59 contributors in 2.062s
npm install eslint-plugin-node@latest --save-dev
npm WARN conventional-changelog-lint-config-cz@0.3.0 requires a peer of conventional-changelog-lint@~1.1.0 but none is installed. You must install peer dependencies yourself.
+ eslint-plugin-node@6.0.1
updated 1 package in 21.437s
npm ERR! Object.entries is not a function
```

ev1stensberg commented:
Which nodejs version are you on? Also, could you rebase against the `next` branch? This PR is quite bad if anything goes wrong :D

PlayMa256 commented:
I'm on 8.11 and also tested on 6.11. I'm going to double test on v6 before rebasing.

PlayMa256 commented:
Also, should this be merged into next too?
